.github/workflows/publish.yml

```diff
@@ -0,0 +1,52 @@
+name: Publish Python Package
+
+on:
+  release:
+    types: [created]
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: pip
+        cache-dependency-path: pyproject.toml
+    - name: Install dependencies
+      run: |
+        pip install '.[test]'
+    - name: Run tests
+      run: |
+        pytest
```

Comment:
why this step is different from the `Run tests` in `test.yml`?

Reply:
I'm not sure why. Both workflows were generated by https://github.com/simonw/python-lib-template-repository.
However, we can remove this step entirely if we don't include pytest in this library.